src/main/java/ui/components/pickers/LabelPickerDialog.java

```diff
@@ -3,157 +3,256 @@
+    }
 
-    private void initialiseDialog(Stage stage, TurboIssue issue) {
-        initOwner(stage);
-        initModality(Modality.APPLICATION_MODAL); // TODO change to NONE for multiple dialogs
-        setTitle("Edit Labels for " + (issue.isPullRequest() ? "PR #" : "Issue #") +
-                issue.getId() + " in " + issue.getRepoId());
+    private final Node processSuggestedLabel(String suggestedLabel) {
+        return getStyledPickerLabel(
+             new PickerLabel(TurboLabel.getFirstMatchingTurboLabel(allLabels, suggestedLabel), true)
+             .faded(true));
+    }
+
+    private final void populateFeedbackLabels(List<String> assignedLabels, List<String> matchedLabels,
+                                        Optional<String> suggestion) {
+        feedbackLabels.getChildren().clear();
+        populateGroupLabels(assignedLabels, matchedLabels, suggestion);
+        populateGrouplessLabels(assignedLabels, matchedLabels, suggestion);
+    }
+
+    private final void populateGroupLabels(List<String> finalLabels, List<String> matchedLabels,
+                                     Optional<String> suggestion) {
+
+        Map<String, FlowPane> groupContent = getGroupContent(finalLabels, matchedLabels, suggestion);
+        groupContent.entrySet().forEach(entry -> {
+            feedbackLabels.getChildren().addAll(
+                    createGroupTitle(entry.getKey()), entry.getValue());
+        });
+    }
+
+    private final Map<String, FlowPane> getGroupContent(List<String> finalLabels, List<String> matchedLabels,
+                                                  Optional<String> suggestion) {
+        Map<String, FlowPane> groupContent = new HashMap<>();
+        allLabels.stream().sorted()
+                .filter(label -> label.isInGroup())
+                .map(label -> new PickerLabel(label, false))
+                .forEach(label -> {
+                    String group = label.getGroupName();
+                    if (!groupContent.containsKey(group)) {
+                        groupContent.put(group, createGroupPane(GROUP_PAD));
+                    }
+                    groupContent.get(group).getChildren().add(processMatchedLabel(
+                        label.getFullName(), matchedLabels, finalLabels, suggestion));
+                });
+        return groupContent;
+    }
+
+    private final void populateGrouplessLabels(List<String> finalLabels, List<String> matchedLabels,
+                                         Optional<String> suggestion) {
+        FlowPane groupless = createGroupPane(GROUPLESS_PAD);
+        allLabels.stream()
+                .filter(label -> !label.isInGroup())
+                .map(label -> new PickerLabel(label, false))
+                .forEach(label -> groupless.getChildren().add(processMatchedLabel(
+                    label.getFullName(), matchedLabels, finalLabels, suggestion)));
+
+        if (!groupless.getChildren().isEmpty()) feedbackLabels.getChildren().add(groupless);
```

Comment:
What happens if you just add without checking for empty result? Still the same result?